samples/apps/copilot-chat-app/webapi/CopilotChat/Skills/ChatSkills/ExternalInformationSkill.cs

```diff
@@ -148,6 +133,46 @@ public async Task<string> AcquireExternalInformationAsync(string userIntent, SKC
 
     #region Private
 
+    /// <summary>
+    /// Scrubs plan of functions not available in Planner's kernel.
+    /// </summary>
+    private Plan SanitizePlan(Plan plan, SKContext context)
+    {
+        List<Plan> sanitizedSteps = new();
+        var availableFunctions = this._planner.Kernel.Skills.GetFunctionsView(true);
+
+        foreach (var step in plan.Steps)
+        {
+            if (this._planner.Kernel.Skills.TryGetFunction(step.SkillName, step.Name, out var function))
+            {
+                this.MergeContextIntoPlan(context.Variables, step.Parameters);
```

Comment:
Does this need to be done? I had thought this would happen automatically on execution.

Reply:
Yeah this is required mainly for the PlanViewer to show the true input values to the user, or else it shows the planner's proposed values since these variables aren't overridden until execution.